sources/engine/Stride.Physics/Engine/PhysicsConstraintProcessor.cs

```diff
@@ -0,0 +1,92 @@
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Stride.Core;
+using Stride.Core.Annotations;
+using Stride.Core.Diagnostics;
+using Stride.Engine;
+using Stride.Games;
+
+namespace Stride.Physics.Engine
+{
+    public class PhysicsConstraintProcessor : EntityProcessor<PhysicsConstraintComponent>
+    {
+        private static readonly Logger logger = GlobalLogger.GetLogger(nameof(PhysicsConstraintProcessor));
+
+        public PhysicsConstraintProcessor()
+        {
+            Order = 0xFFFE; // Before PhysicsProcessor
+        }
+
+        protected override void OnEntityComponentAdding(Entity entity, [NotNull] PhysicsConstraintComponent component, [NotNull] PhysicsConstraintComponent data)
+        {
+            // this is mostly required for the game studio gizmos
+            if (Simulation.DisableSimulation)
+            {
+                return;
+            }
+
+            Recreate(component);
+        }
+
+        protected override void OnEntityComponentRemoved(Entity entity, [NotNull] PhysicsConstraintComponent component, [NotNull] PhysicsConstraintComponent data)
+        {
+            if (component.Constraint != null)
+            {
+                component.Simulation.RemoveConstraint(component.Constraint);
+                component.Constraint = null;
+                // see RigidbodyComponent.OnDetach about disposing constraints
+            }
+        }
+
+        public override void Update(GameTime time)
+        {
+            // this is mostly required for the game studio gizmos
+            if (Simulation.DisableSimulation)
+                return;
+
+            foreach (var datas in ComponentDatas)
+            {
+                var component = datas.Key;
+
+                if (component.Constraint != null)
+                {
+                    component.Constraint.Enabled = component.Enabled;
+                }
+                else if (component.Constraint == null && component.Enabled)
+                {
+                    Recreate(component);
+                }
+            }
+        }
+
+        public void Recreate(PhysicsConstraintComponent component)
+        {
+            if (component.Constraint != null)
+            {
+                component.Simulation.RemoveConstraint(component.Constraint);
+            }
+            
+            if (component.Enabled)
+            {
+                if (component.Description == null || component.BodyA == null)
+                {
+                    logger.Warning("ConstraintComponent with an empty description or missing required body. Skipping constraint creation.");
+                    return;
+                }
+
+                if (component.BodyB != null && component.BodyB.Simulation != component.BodyA.Simulation)
+                    return; // simulation mismatch - may happen when first loading the scene
```

Comment:
Shouldn't this case and the one below throw then since the function could not recreate it ? Or is it going to be recreated automatically later in the frame ?

Reply:
I need to check again what was up with that - I can't remember the exact reason why this processor was running before the physics one (I remember there was some something, but not the details) - it's possible that it should be the other way around - thus ensuring rigidbodies are properly initialized by the time this runs.

Reply:
So apparently the ordering of adding entities' components is a little different than I expected (e.g. child entity is added in the transform component processor, so their components will be processed before any other parent components). I think I did a good job mitigating it (see new commit).

The one thing that bugs me a little is - if you remove a rigidbody before you remove the constraint component, on the next frame the constraint component will try to recreate itself and throw an exception because the rigidbody is no longer valid. Do you think I should be handling this in some way - i.e. set some flag on the component not to recreate the constraint unless user explicitly calls `Recreate()` or removes and re-adds the component, or just have them ensure that they remove the constraint component on the same frame as they remove the rigid body?

Reply:
I see, perhaps the ``PhysicsConstraintComponent`` should be a property of the ``RigidbodyComponent`` then if the former always requires the later and creates unexpected issues when it is removed ?

Then you would just need to release the constraint when ``BodyB`` was set but is no longer in the scene ?

Reply:
Well, my approach to constraints has always been a little to the side of rigidbodies - which is why I didn't tie `BodyA` to the entity holding the component - this allowed me to put all constraints into a single entity when doing a ragdoll, which made it a bit easier to place constraints for left/right side, rather than switching between entities and trying to memorize the placements of constraint pivots. Plus, I feel like the `RigidbodyComponent` already has a lot going on.

I see two options:
- one is to provide an event on the constraint component `RigidbodyDetach`, which would be called when one of the rigidbodies gets detached - this can be then wired up by the users with a callback, but would likely require people to set up those components from code
- add an internal flag `Detached`, which would prevent exceptions when a rigidbody referenced in the constraint was previously successfully in a constraint but then was detached, but should the rigidbody be reattached the constraint would be re-established.

And document the behaviour so that users aren't surprised.

Reply:
Sure, that makes sense.
I don't think I would expect constraints to throw, and in the small number of cases where I need to ensure that those connections will not break, a callback seems like the best tool for the job so, I would rather go for the first option.
Release the connection as soon as either body is no longer in the scene or if they are set to null by the user and invoke a method to let them know, yeah.

Reply:
@Eideren have you had a chance to look at the last update?